[Yuki Akizuki] iP

yuki-cell opened the pull request.

Commit:
Add Gradle support

yuki-cell committed:
commit Level-1

yuki-cell committed:
commit level-2

yuki-cell committed:
commit Level 3

yuki-cell committed:
commit Level-4

yuki-cell committed:
used A-TextUiTesting

yuki-cell committed:
commit Level-5

yuki-cell committed:
commit Level-6

yuki-cell committed:
commit Level-7

yuki-cell committed:
commit branch Level-8

yuki-cell committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	iP/out/production/iP/Duke.class
#	iP/src/Duke.java

yuki-cell committed:
commit A-MoreOOP

yuki-cell committed:
commit A-Packages

yuki-cell committed:
commit A-JUnit

yuki-cell committed:
commit A-JUnit(newer)

yuki-cell committed:
commit A-Jar

yuki-cell committed:
commit A-JavaDoc

yuki-cell committed:
commit A-CodingStandard

yuki-cell committed:
commit branch Level-9

yuki-cell committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	iP/src/duke/Ui.java

YangJiyu98 commented on iP/src/duke/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+package duke;
```
Maybe can consider having a empty new line to group your package/import statements

gordonfgz replied:
I agree

yongping827 replied:
I second the suggestion made by @YangJiyu98

YangJiyu98 commented on iP/src/duke/Duke.java:
```diff
@@ -0,0 +1,172 @@
+
+        } else {
+            // invalid input
+            if (!loading){
```
You might have missed a whitespace here

gordonfgz replied:
I agree

yongping827 replied:
I second that

YangJiyu98 commented on iP/src/duke/Duke.java:
```diff
@@ -0,0 +1,172 @@
+                ui.showError(e.getMessage());
+            }
+
+        } else if (instructionType.equals("event")){
```
You might have missed a white space here

gordonfgz replied:
I agree

yongping827 replied:
I second that

YangJiyu98 commented on iP/src/duke/Duke.java:
```diff
@@ -0,0 +1,172 @@
+        }
+    }
+
+    /**
+     * run todo app
+     */
```
Not sure if this would be necessary as it is simple enough fro a reader to understand

YangJiyu98 reviewed (commented):
Very neat code! very easy to read

yuki-cell committed:
Merge commit '3b19ba1790d6035a9e246f5d4ae627bf68778606'

yeohhq commented on iP/src/duke/Duke.java:
```diff
@@ -1,6 +1,7 @@
 package duke;
```
Maybe you can have a new empty line indentation after your package line before your import statements?

gordonfgz replied:
I agree

yongping827 replied:
Repeated suggestion. I second this as well.

yeohhq commented on iP/src/duke/TaskList.java:
```diff
@@ -47,4 +47,14 @@ public Task getTask(int index) {
         return task;
     }
+
+    public ArrayList<Task> find(String keyword) {
```
I like how you return your output as an ArrayList<Task> to be processed later instead of immediately returning a processed string since this could improve adaptability and extensibility of your program! 👍

gordonfgz replied:
I agree

yeohhq commented on iP/src/duke/Ui.java:
```diff
@@ -107,4 +107,14 @@ public String readCommand() {
+
+    public void showFoundTask(ArrayList<Task> foundTasks) {
+        System.out.println("Here are the matching tasks in your list:");
+        int index = 1;
```
Perhaps it would be more readable if you had the for-loop in the form of:
for (int i = 1; i < foundTasks.size() + 1; i++) ?

gordonfgz replied:
I think that the current way is fine and I actually like it more

yeohhq reviewed (commented):
I think your code is generally easy to read and follow, with inline comments to also improve readability which is good! :)

gordonfgz commented on iP/src/duke/Duke.java:
```diff
@@ -0,0 +1,172 @@
+        try {
+            // need empty tasks to load properly
+            tasks = new TaskList();
+            tasks = new TaskList(storage.load(this));
```
Perhaps there is a way of changing your code design so that you can initialise with just one line of code instead of calling the constructor twice?

gordonfgz commented on iP/src/duke/Duke.java:
```diff
@@ -0,0 +1,172 @@
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
```
Perhaps you can put your comment on top of the line of code instead of beside it?
Not really sure if this violates Java coding standard.

gordonfgz commented on iP/src/duke/Duke.java:
```diff
@@ -0,0 +1,172 @@
+        // save Data for every user input
+        Storage.save(tasks.taskList);
+    }
+}
```
Should this be extracted out? Any reasons why you chose to have the input processing method in Duke.java instead of a separate class?

gordonfgz commented on iP/src/duke/Task.java:
```diff
@@ -0,0 +1,42 @@
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
I like how you named this member. It is very clear that its going to be a boolean!

yuki-cell committed:
commit A-Gradle

yongping827 commented on iP/src/duke/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+    protected LocalDate deadline;
+
+    /**
+     * constructor of Deadline
```
Perhaps capitalize the beginning of JavaDoc comments? This applies to all other JavaDoc comments as well.

yongping827 commented on iP/src/duke/Duke.java:
```diff
@@ -0,0 +1,172 @@
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
```
Perhaps group imports based on the packages and leave an empty line between groups?

yongping827 commented on iP/src/duke/Duke.java:
```diff
@@ -0,0 +1,172 @@
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+                 */
```
Not part of the Java coding standard, but perhaps remove commented code that I would presume is unwanted?

yongping827 reviewed (commented):
I would suggest to gitignore build files, i.e. in this context, the directories `iP/.idea` and `iP/out`. The reason behind this is any changes to the code would change the files in these directories, and would create too many unnecessary changes in the git history.

jianhandev commented on iP/src/duke/Duke.java:
```diff
@@ -0,0 +1,172 @@
+     * @param filePath path where saved data of todo is stored
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            // need empty tasks to load properly
```
From the java coding standards: "Logical units within a block should be separated by one blank line."
Consider leaving a line before the try block?

jianhandev commented on iP/src/duke/Duke.java:
```diff
@@ -0,0 +1,172 @@
+                // parse instruction
+                HashMap<String, Object> parsedData = Parser.parseAddEventInstr(user_input);
+                String description = (String) parsedData.get("description");
+                LocalDate l_time = (LocalDate) parsedData.get("time");
```
Variable name could be more descriptive.

jianhandev commented on iP/src/duke/Parser.java:
```diff
@@ -0,0 +1,108 @@
+     * @param user_input input give from user
+     * @return hashMap containing description and deadline of task
+     */
+    static HashMap<String, Object> parseAddDeadlineInstr(String user_input) throws DukeException {
+        // check if input is valid
+        if (user_input.split(" ", 2).length == 1) {
+            throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+        }
+        String task = user_input.split(" ", 2)[1];
+        if (task.split(" /by ", 2).length < 2) {
+            throw new DukeException("☹ OOPS!!! The description and time is required for deadline");
+        }
+        // get data
+        String description = task.split(" /by ", 2)[0];
+        String time = task.split(" /by ")[1];
+        LocalDate l_time = LocalDate.parse(time);
+        HashMap<String, Object> parsedData = new HashMap<String, Object>(){
+            { put("description", description); put("time", l_time); }
+        };
+        return parsedData;
+    }
```
This block is generally abit chaotic, perhaps add in line breaks to separate different logical units?

jianhandev commented on iP/src/duke/Ui.java:
```diff
@@ -0,0 +1,120 @@
+    }
+
+    /**
+     * show welcome message when app is started
```
Javadocs:
- "show" should be "Shows"
- add a full stop after the function description

yuki-cell committed:
commit Level-10

yuki-cell committed:
commit FXML implementation of GUI

yuki-cell committed:
commit change in root folder
before: root = CS2103 > iP(redundant folder) > folders
after: root = CS2103 > folders

yuki-cell committed:
commit A-Assertions

yuki-cell committed:
commit A-CodingQuality

yuki-cell committed:
Merge pull request #2 from yuki-cell/branch-A-Assertions

Merge A-Assertions branch to master branch

yuki-cell committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java

yuki-cell committed:
Merge pull request #3 from yuki-cell/branch-A-CodeQuality

Branch a code quality branch to master branch

yuki-cell committed:
commit BCD-Extension (C-Priority)

yuki-cell committed:
commit A-UserGuide

yuki-cell committed:
Set theme jekyll-theme-cayman

yuki-cell committed:
commit A-Release

yuki-cell committed:
Merge branch 'master' of https://github.com/yuki-cell/ip

yuki-cell committed:
commit split UserInputHandler() function to multiple Command class

yuki-cell committed:
commit test update of user guide website

yuki-cell committed:
Commit update user guide

yuki-cell committed:
Commit update user guide

yuki-cell committed:
commit update user guide

yuki-cell committed:
commit update user guide

yuki-cell committed:
commit update user guide

yuki-cell committed:
commit update user guide

yuki-cell committed:
commit update user guide

yuki-cell committed:
commit update user guide

yuki-cell committed:
commit update user guide

yuki-cell committed:
commit update user guide

yuki-cell committed:
commit update user guide

yuki-cell committed:
commit fix minor issues

yuki-cell committed:
commit fix minor issues
